woocommerce/woocommerce-android
# IAP: Showcase

Closes: #7514


### Description

The PR adds an activity (available only in debug) that can be used to showcase/test the IAP library

Notice, that information from the EditText is not used at the moment, because the API calls are yet to be implemented. At the moment the library communicates only with IAP service

It allows
* Fetch plan details (now it's a subscription created for testing purposes)
* Check if IAP supported (at the moment it checks if the currency a user will use to buy it is USD)
* Check if plans has been purchased already
* Make a purchase

### Testing instructions


The easiest way to start the activity is to use the following configuration:

<img width="1097" alt="image" src="https://user-images.githubusercontent.com/4923871/196628912-c67d07e9-cc31-4af2-8365-7b2e690bb4e7.png">

Also, make sure to use `vanillaDebug` build variant, otherwise package name won't match and IAP service won't work with the app

* Make sure you use a google account that is part of the license testing (I could not get permission to edit that, so probably Cesar has to be asked to add you to the list because Jirka is afk)
* Try to click all 4 buttons and make sure that all of them do what's written on them (take a look into attached video)


// cc @0nko you probably have a lot to do right now, but if you'll give a quick look into [the view model](https://github.com/woocommerce/woocommerce-android/blob/c4cc4daaf0a579fc63785b240dd9783c2fd5b67e/WooCommerce/src/debug/kotlin/com.woocommerce.android/iapshowcase/IAPShowcaseViewModel.kt) it might give you some understanding of how the library can be used. You might not need, to use `isWPComPlanPurchased` method. It depends on the flow

### Images/gif


https://user-images.githubusercontent.com/4923871/196203984-905f02a9-8f0c-49ff-8e0a-5dbd2160fa0a.mp4




### We are going to iterate, so consider this code as the first iteration. It doesn't go to the trunk and will be reviewed/tested again
